core/src/identity.rs

```diff
@@ -33,11 +33,14 @@
 //! All key types have functions to enable conversion to/from their binary representations.
 
 #[cfg(feature = "ecdsa")]
+#[cfg_attr(docsrs, doc(cfg(feature = "ecdsa")))]
```

Comment:
I wonder whether it is worth it using a macro to remove this duplication. I can see a couple of downsides, not sure how critical they actually are:

- Less flexibility in specifying the rendered feature combination.
- Increased compile-time because the macro needs to run before all our other crates.
- Newcomers will be unfamiliar with the solution.

Reply:
I just found `doc_auto_cfg` which seems to be useful to automate all of this. The magic line seems to be:

```rust
#![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
```

See https://github.com/tokio-rs/valuable/pull/80/files for example.

Reply:
I will try that, although if we decide to go with this, it would be much easier for me to do that in a new branch (and pull request).